File: BuilderDesignPattern/CarManufacture_Example/Products/CarManual.java
package BuilderDesignPattern.CarManufacture_Example.Products;

import BuilderDesignPattern.CarManufacture_Example.Components.Engine;
import BuilderDesignPattern.CarManufacture_Example.Components.GPSNavigator;
import BuilderDesignPattern.CarManufacture_Example.Components.Transmission;
import BuilderDesignPattern.CarManufacture_Example.Components.TripComputer;
import BuilderDesignPattern.CarManufacture_Example.ProductTypes.CarsType;

public class CarManual extends Product {
	private final String model;
	private final CarsType carType;
	private final int seats;
	private final Engine engine;
	private final Transmission transmission;
	private final TripComputer tripComputer;
	private final GPSNavigator gpsNavigator;

	/**
	 * @param carModel
	 * string name of the car manual.
	 * @param carType
	 * enum for the type of car manual.
	 * @param seats
	 * int number of seats of car manual.
	 * @param engine
	 * engine for the engine of car manual.
	 * @param transmission
	 * transmission for the transmission of car manual.
	 * @param tripComputer
	 * tripComputer for the tripComputer of car manual.
	 * @param gpsNavigator
	 * gpsNavigator for the gpsNavigator of car manual.
	 */
	public CarManual(String carModel, CarsType carType, int seats, Engine engine, Transmission transmission, TripComputer tripComputer, GPSNavigator gpsNavigator) {
		this.model  = carModel;
		this.carType = carType;
		this.seats = seats;
		this.engine = engine;
		this.tripComputer = tripComputer;
		this.gpsNavigator = gpsNavigator;
		this.transmission = transmission;
	}

	public String print() {
		String info = "\nCar Manual \n";
		info += "Name of car: " + model + "\n";
		info += "Type of car: " + carType + "\n";
		info += "Count of seats: " + seats + "\n";
		info += "Engine: volume - " + engine.getVolume() + "; mileage - " + engine.getMileage() + "\n";
		info += "Transmission: " + transmission + "\n";
		if (this.tripComputer != null) {
			info += "Trip Computer: Functional" + "\n";
		} else {
			info += "Trip Computer: N/A" + "\n";
		}
		if (this.gpsNavigator != null) {
			info += "GPS Navigator: Functional" + "\n";
		} else {
			info += "GPS Navigator: N/A" + "\n";
		}
		return info;
	}
}

File: BuilderDesignPattern/CarManufacture_Example_Python/Products/CarManual.py
from BuilderDesignPattern.CarManufacture_Example_Python.Components.Engine import Engine
from BuilderDesignPattern.CarManufacture_Example_Python.Components.GPSNavigator import GPSNavigator
from BuilderDesignPattern.CarManufacture_Example_Python.Components.Transmission import Transmission
from BuilderDesignPattern.CarManufacture_Example_Python.Components.TripComputer import TripComputer
from BuilderDesignPattern.CarManufacture_Example_Python.Products.Product import Product
from BuilderDesignPattern.CarManufacture_Example_Python.ProductsTypes.CarsType import CarsType


class CarManual(Product):
	_model: str
	_type: CarsType
	_seat: int
	_engine: Engine
	_transmission: Transmission
	_tripComputer: TripComputer
	_gpsNavigator: GPSNavigator

	def __init__(self, model: str, carType: CarsType, seat: int, engine: Engine, transmission: Transmission, tripComputer: TripComputer, gpsNavigator: GPSNavigator):
		self._model = model
		self._type = carType
		self._seat = seat
		self._engine = engine
		self._transmission = transmission
		self._tripComputer = tripComputer
		self._gpsNavigator = gpsNavigator

	def __str__(self) -> str:
		trip_comp = "Functional" if self._tripComputer is not None else "N/A"
		gps_nav = "Functional" if self._gpsNavigator is not None else "N/A"

		return (
			f"\nCar Manual\n"
			f"Name of car: {self._model}\n"
			f"Type of car: {self._type}\n"
			f"Count of seats: {self._seat}\n"
			f"Engine: volume - {self._engine.getVolume()}; mileage - {self._engine.getMileage()}\n"
			f"Transmission: {self._transmission}\n"
			f"Trip Computer: {trip_comp}\n"
			f"GPS Navigator: {gps_nav}\n"
		)